fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java

```diff
@@ -87,14 +87,25 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
             }
         }
 
-        if (updateStmt.getWherePredicate() == null) {
-            if (checkIfUsePartialUpdate(assignmentList.size(), table.getBaseSchema().size())) {
-                // use partial update if:
-                // 1. Columns updated are less than 4
-                // 2. The proportion of columns updated is less than 30%
-                // 3. No where predicate in update stmt
-                updateStmt.setUsePartialUpdate();
-            } else {
+        if (session.getSessionVariable().getPartialUpdateMode().equals("column")) {
```

Comment:
column, auto and row should be enum or const?

Reply:
SessionVar not support enum, so add `PartialUpdateModeConverter` to check if it is column, auto or row.